src/common.rs

```diff
@@ -4,22 +4,28 @@ use std::path::PathBuf;
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct Warning {
+    pub check_name: String,
     line: LineEntry,
     message: String,
 }
 
 impl Warning {
-    pub fn new(line: LineEntry, message: String) -> Self {
-        Self { line, message }
+    pub fn new(line: LineEntry, check_name: &str, message: String) -> Self {
+        let check_name = String::from(check_name);
+        Self {
+            line,
+            check_name,
```

Comment:
```suggestion
            check_name: String::from(check_name),
```

It's seems more readable for me, what do you think?

Reply:
I think it is more idiomatic to use only the key in languages ​​that support this feature (like Rust).
But in this case it is not really matter, if it is more readable for most people - I will gladly change it.